airguardian-core/drone/src/main/kotlin/systems/ajax/motrechko/airguardian/drone/infrastructure/adapters/repository/redis/RedisDroneRepository.kt

```diff
@@ -52,18 +64,23 @@ class DroneRedisRepository(
             }
     }
 
-    override fun deleteById(id: String): Mono<DeleteResult> =
-        reactiveRedisRepository
+    override fun deleteById(id: String): Mono<Boolean> {
+        return reactiveRedisRepository
             .delete(dronePrefix + id)
             .then(droneRepository.deleteById(id))
+    }
 
     private fun saveDroneToRedisCache(drone: Drone): Mono<Drone> {
-        val key = drone.id.toHexString()
-        return reactiveRedisRepository.opsForValue()
+        val key = drone.id
+        return if (key != null) {
+            reactiveRedisRepository.opsForValue()
                 .set(
                     dronePrefix + key,
-                    drone,
+                    drone.toMongoDrone(),
```

Comment:
Storing a drone in redis: 
  `Drone` (domain) -> MongoDrone -> store

Find in redis:
 storage -> MongoDrone -> Drone (domain) 

Looks like it should be reworked :)